<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PagerDemodGUI</class>
 <widget class="RollupWidget" name="PagerDemodGUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>404</width>
    <height>764</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>352</width>
    <height>0</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>Liberation Sans</family>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Pager Demodulator</string>
  </property>
  <property name="statusTip">
   <string>Pager Demodulator</string>
  </property>
  <widget class="QWidget" name="settingsContainer" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>390</width>
     <height>171</height>
    </rect>
   </property>
   <property name="minimumSize">
    <size>
     <width>350</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Settings</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>3</number>
    </property>
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="powLayout">
      <property name="topMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLabel" name="deltaFrequencyLabel">
        <property name="minimumSize">
         <size>
          <width>16</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Df</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ValueDialZ" name="deltaFrequency" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>32</width>
          <height>16</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Liberation Mono</family>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
        <property name="toolTip">
         <string>Demod shift frequency from center in Hz</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="deltaUnits">
        <property name="text">
         <string>Hz </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="channelPowerLayout">
        <item>
         <widget class="QLabel" name="channelPower">
          <property name="toolTip">
           <string>Channel power</string>
          </property>
          <property name="layoutDirection">
           <enum>Qt::RightToLeft</enum>
          </property>
          <property name="text">
           <string>0.0</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="channelPowerUnits">
          <property name="text">
           <string> dB</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="powerLayout">
      <item>
       <widget class="QLabel" name="channelPowerMeterUnits">
        <property name="text">
         <string>dB</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="LevelMeterSignalDB" name="channelPowerMeter" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Liberation Mono</family>
          <pointsize>8</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>Level meter (dB) top trace: average, bottom trace: instantaneous peak, tip: peak hold</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_5">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="phySettingsLayout">
      <item>
       <widget class="QLabel" name="rfBWLabel">
        <property name="text">
         <string>BW</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="rfBW">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>RF bandwidth</string>
        </property>
        <property name="minimum">
         <number>10</number>
        </property>
        <property name="maximum">
         <number>400</number>
        </property>
        <property name="pageStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>100</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="rfBWText">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>10.0k</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="fmDevLabel">
        <property name="text">
         <string>Dev</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="fmDev">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Frequency deviation</string>
        </property>
        <property name="minimum">
         <number>10</number>
        </property>
        <property name="maximum">
         <number>50</number>
        </property>
        <property name="pageStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>24</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="fmDevText">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>2.4k</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="phySettingsLayout_2">
      <item>
       <widget class="QLabel" name="modulationLabel">
        <property name="text">
         <string>Mod</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="modulation">
        <property name="minimumSize">
         <size>
          <width>76</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Pager modulation</string>
        </property>
        <item>
         <property name="text">
          <string>POCSAC</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_10">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="baudLabel">
        <property name="text">
         <string>Baud</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="baud">
        <property name="minimumSize">
         <size>
          <width>60</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Baud rate</string>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <item>
         <property name="text">
          <string>512</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1200</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>2400</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line_11">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="decodeLabel">
        <property name="text">
         <string>Decode</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="decode">
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>0</height>
         </size>
        </property>
        <property name="toolTip">
         <string>How to choose between decoding a message as numeric or alphanumeric</string>
        </property>
        <item>
         <property name="text">
          <string>Standard</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Inverted</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Numeric</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Alphanumeric</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Heuristic</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="charset">
        <property name="toolTip">
         <string>Select character encoding</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/keyboard.png</normaloff>:/keyboard.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_6">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="udpLayout">
      <item>
       <widget class="QCheckBox" name="udpEnabled">
        <property name="toolTip">
         <string>Forward messages via UDP</string>
        </property>
        <property name="layoutDirection">
         <enum>Qt::RightToLeft</enum>
        </property>
        <property name="text">
         <string>UDP</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="udpAddress">
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>0</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="toolTip">
         <string>Destination UDP address</string>
        </property>
        <property name="inputMask">
         <string>000.000.000.000</string>
        </property>
        <property name="text">
         <string>127.0.0.1</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="udpSeparator">
        <property name="text">
         <string>:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="udpPort">
        <property name="minimumSize">
         <size>
          <width>50</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>50</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::ClickFocus</enum>
        </property>
        <property name="toolTip">
         <string>Destination UDP port</string>
        </property>
        <property name="inputMask">
         <string>00000</string>
        </property>
        <property name="text">
         <string>9998</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="filterLayout">
      <item>
       <widget class="QLabel" name="filterAddressLabel">
        <property name="text">
         <string>Find</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="filterAddress">
        <property name="toolTip">
         <string>Display only messages where the address matches the specified regular expression</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="clearTable">
        <property name="toolTip">
         <string>Clear messages from table</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../../../sdrgui/resources/res.qrc">
          <normaloff>:/bin.png</normaloff>:/bin.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="messageContainer" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>210</y>
     <width>391</width>
     <height>171</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="windowTitle">
    <string>Received Messages</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayoutTable">
    <property name="spacing">
     <number>2</number>
    </property>
    <property name="leftMargin">
     <number>3</number>
    </property>
    <property name="topMargin">
     <number>3</number>
    </property>
    <property name="rightMargin">
     <number>3</number>
    </property>
    <property name="bottomMargin">
     <number>3</number>
    </property>
    <item>
     <widget class="QTableWidget" name="messages">
      <property name="toolTip">
       <string>Received messages</string>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <column>
       <property name="text">
        <string>Date</string>
       </property>
       <property name="toolTip">
        <string>Date message was received</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Time</string>
       </property>
       <property name="toolTip">
        <string>Time message was received</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Address</string>
       </property>
       <property name="toolTip">
        <string>21-bit pager address the message is for</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Message</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Function</string>
       </property>
       <property name="toolTip">
        <string>Function bits / source address</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Alpha</string>
       </property>
       <property name="toolTip">
        <string>Message decoded as 7-bit alphanumeric characters</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Numeric</string>
       </property>
       <property name="toolTip">
        <string>Message decoded as 4-bit BCD numeric characters</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Even PE</string>
       </property>
       <property name="toolTip">
        <string>Number of even parity errors detected in message</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>BCH PE</string>
       </property>
       <property name="toolTip">
        <string>Number of BCH parity errors detected in message</string>
       </property>
      </column>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="scopeContainer" native="true">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>400</y>
     <width>351</width>
     <height>341</height>
    </rect>
   </property>
   <property name="windowTitle">
    <string>Waveforms</string>
   </property>
   <layout class="QVBoxLayout" name="transmittedLayout_2">
    <property name="spacing">
     <number>2</number>
    </property>
    <property name="leftMargin">
     <number>3</number>
    </property>
    <property name="topMargin">
     <number>3</number>
    </property>
    <property name="rightMargin">
     <number>3</number>
    </property>
    <property name="bottomMargin">
     <number>3</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="scopelLayout">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Real</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="channel1">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Signal to send to scope as real part</string>
        </property>
        <item>
         <property name="text">
          <string>I</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Q</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Mag Sq</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>FM demod</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Filt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>DC offset</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Data</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sample</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Bit</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GotSyncCode</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Imag</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="channel2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Signal to send to scope as imag part</string>
        </property>
        <item>
         <property name="text">
          <string>I</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Q</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Mag Sq</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>FM demod</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Filt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>DC offset</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Data</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sample</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Bit</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GotSyncCode</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="GLScope" name="glScope" native="true">
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>250</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Liberation Mono</family>
        <pointsize>8</pointsize>
       </font>
      </property>
     </widget>
    </item>
    <item>
     <widget class="GLScopeGUI" name="scopeGUI" native="true"/>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>RollupWidget</class>
   <extends>QWidget</extends>
   <header>gui/rollupwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ValueDialZ</class>
   <extends>QWidget</extends>
   <header>gui/valuedialz.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LevelMeterSignalDB</class>
   <extends>QWidget</extends>
   <header>gui/levelmeter.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GLScope</class>
   <extends>QWidget</extends>
   <header>gui/glscope.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>GLScopeGUI</class>
   <extends>QWidget</extends>
   <header>gui/glscopegui.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../sdrgui/resources/res.qrc"/>
 </resources>
 <connections/>
</ui>
